### ISSUE TITLE
Stats collection

### ISSUE THREAD

**Issue by IdoitEngineer** [irvined1982/lavaflow#11]
I've finally managed to get the lava-web & lavaFlow to generate pages.
1) How does lavaFlow collect openlava stats? Is there a cron job I need to run to build the stats?
2) The cmd line python scripts  ... Where the docs say URL; does that mean the fastcgi server?

**Comment by irvined1982** [comment]
Hi Jeringa,

Thanks for trying out Lavaflow, I haven't published the docs online yet, (Currently writing them.) but in the doc directory there are some html pages that document in more detail.  They should progress fairly quickly over the coming weeks as I'm doing a lot of tidying up on the UI, docs, and DB.  

What I do to import data is this:

manually run ./import/lava-import-openlava.py on lsb.events\* and lsb.acct\* once to import all historical data

Then, (i haven't got round to writing an init script for this yet) run ./import/lava-import-openlava.py with the --tail_log option - this will keep reading the log file, and upload new data after --chunk_size entries.

URL should be the URL of the lavaFlow instance on the fastCGI server, so lets say the mail page for lavaflow is http://yourserver.com/lavaFlow then you would use that as the URL.

Hope that helps, let me know how you get on

David

**Comment by IdoitEngineer** [comment]
Getting there!
but after importing both the lsb.events and the lsb.acct files I end up with
ZeroDivisionError at /lavaFlow/users/
float division by zero
Request Method: GET
Request URL:    http://testhost:8888/test_olweb/lavaFlow/users/
Django Version: 1.6.5
Exception Type: ZeroDivisionError
Exception Value:  
float division by zero
Exception Location: /opt/lavaweb/lib/python2.7/site-packages/django_lavaflow-1.1-py2.7.egg/lavaFlow/models.py in average_pend_time_percent, line 840
Python Executable:  /opt/lavaweb/bin/python
Python Version: 2.7.2

**Comment by IdoitEngineer** [comment]
Error during template rendering

In template /opt/.../templates/lavaFlow/widgets/user_list.html, error at line 28
float division by zero

```
18              <tr>
19                  <td><a href='{{ object.get_absolute_url }}'>{{ object.name }}</a></td>
20                  <td>{{ object.total_jobs }}</td>
21                  <td>{{ object.total_tasks }}</td>
22                  <td>{{ object.total_attempts }}</td>
23                  <td><a href="{{ object.last_task.job.get_absolute_url }}">{{ object.last_task.job.submit_time_datetime }}</a></td>
24                  <td><a href="{{ object.last_task.get_absolute_url }}">{{ object.last_task.end_time_datetime }}</a></td>
25                  <td><a href="{{ object.last_failed_task.get_absolute_url }}">{{ object.last_failed_task.end_time_datetime }}</a></td>
26                  <td>{{ object.average_pend_time_timedelta }}</td>
27                  <td>{{ object.average_wall_time_timedelta }}</td>
28                      <td {% if object.average_pend_time_percent > 50 %} class="danger"{% elif object.average_pend_time_percent > 25 %}class="warning"{% endif %}>{{ object.average_pend_time_percent|floatformat:"2" }}</td>
29              </tr>
30              {% empty %}
31              <tr><td colspan="10">No Users.</td></tr>
32              {% endfor %}
33          </tbody>
34      </table>
```

It's moaning about line 28 where it has 

```
{{ object.average_pend_time_percent|floatformat:"2" }}
```

**Comment by irvined1982** [comment]
Hi Jerenga,

Sorry about that - I'm guessing this user had cancelled all jobs before they started executing, this is now fixed for this case (and projects/clusters/etc) (Commit: 2615d3a81476211374fd3ac7d57e097495843b88)

Depending on when you last did a git pull, you may need to do update the db - if you see a bunch of exceptions relating to database fields, you'll need to clear the db, sync it, and re-import.
The following one liner will take care of the DB for you and recreate a key.

python manage.py sqlclear lavaFlow|python manage.py dbshell; python manage.py syncdb; python manage.py shell <<EOF
from lavaFlow.models import ImportKey
ImportKey.models.get_or_create(client_key="MyKey")
EOF

**Comment by IdoitEngineer** [comment]
KeyError at /lavaFlow/0/0/exclude/none/exclude/filter/user__id__in.1/filter/group/none/group
u'user__id'
Request Method: GET
Request URL:    http://testserver:8888/test_olweb/lavaFlow/0/0/exclude/none/exclude/filter/user__id__in.1/filter/group/none/group
Django Version: 1.6.5
Exception Type: KeyError
Exception Value:  
u'user__id'
Exception Location: /opt/lavaweb/lib/python2.7/site-packages/django_lavaflow-1.1-py2.7.egg/lavaFlow/views.py in set_filters, line 1803
Python Executable:  /opt/lavaweb/bin/python
Python Version: 2.7.2
Python Path:  
['/opt/test_olweb',
 '/opt/lavaweb/lib/python2.7/site-packages/setuptools-0.6c11-py2.7.egg',
 '/opt/lavaweb/lib/python2.7/site-packages/pip-1.2.1-py2.7.egg',
 '/opt/lavaweb/lib/python2.7/site-packages/django_openlavaweb-1.0-py2.7.egg',
 '/opt/lavaweb/lib/python2.7/site-packages/django_lavaflow-1.1-py2.7.egg',
 '/opt/lavaweb/lib/python2.7/site-packages/OpenlavaWeb_clients-0.0.1-py2.7.egg',
 '/opt/lavaweb/lib/python27.zip',
 '/opt/lavaweb/lib/python2.7',
 '/opt/lavaweb/lib/python2.7/plat-linux2',
 '/opt/lavaweb/lib/python2.7/lib-tk',
 '/opt/lavaweb/lib/python2.7/lib-old',
 '/opt/lavaweb/lib/python2.7/lib-dynload',
 '/software/Python-2.7.2/lib/python2.7',
 '/software/Python-2.7.2/lib/python2.7/plat-linux2',
 '/software/Python-2.7.2/lib/python2.7/lib-tk',
 '/opt/lavaweb/lib/python2.7/site-packages']
Server time:    Mon, 11 Aug 2014 23:51:34 +0000

I've recreated the DB, Tried removing re cloning all the lavaFlow, openlava-python & openlava-web stuff. 

python manage.py syncdb runs fine without errors
manage.py shell <<EOF
from lavaFlow.models import ImportKey
ImportKey.models.get_or_create(client_key="user" )
       gives
AttributeError: type object 'ImportKey' has no attribute 'models'

Sorry if I'm being dense ... But I'm not sure what to try next or how to find out what client_key should be

**Comment by irvined1982** [comment]
Jerenga,

Not dense at all, I appreciate your patience, and help chasing down these bugs.  I inadvertantly broke a few things when I updated the UI, and one of them was the URLs for the users view (again),  It's fixed in: 5958e1d2aad53887fcc80d121e8d1e5ff226f49f along with a date issue on that view too.  

There's no need to update the database again for this.  You can set a client key via the admin interface, or through the python shell, I made a typo, to do it via the shell, you want this:

manage.py shell <<EOF
from lavaFlow.models import ImportKey
ImportKey.objects.get_or_create(client_key=XXXXXX )

Where XXXXXX is an arbitrary string is a pre shared key that will be used to authenticate that the import script is allowed to upload data.  

Hope that helps, and good luck!
